src/libraries/Common/tests/System/IO/Compression/ZipTestHelper.cs

```diff
@@ -336,9 +336,20 @@ public static async Task DirsEqual(string actual, string expected)
 
         public static void DirFileNamesEqual(string actual, string expected)
         {
-            IOrderedEnumerable<string> actualEntries = Directory.EnumerateFileSystemEntries(actual, "*", SearchOption.AllDirectories).Order();
-            IOrderedEnumerable<string> expectedEntries = Directory.EnumerateFileSystemEntries(expected, "*", SearchOption.AllDirectories).Order();
-            AssertExtensions.SequenceEqual(expectedEntries.Select(Path.GetFileName).ToArray(), actualEntries.Select(Path.GetFileName).ToArray());
+            IEnumerable<string> actualEntries = Directory.EnumerateFileSystemEntries(actual, "*", SearchOption.AllDirectories);
+            IEnumerable<string> expectedEntries = Directory.EnumerateFileSystemEntries(expected, "*", SearchOption.AllDirectories);
+
+            if (PlatformDetection.IstvOS || PlatformDetection.IsiOS)
+            {
+                actualEntries = actualEntries.Select(Path.GetFileName).Select(name => name.Normalize(NormalizationForm.FormC)).Order();
+            }
```

Comment:
I think there should be a comment as to why we are treating these differently than the others.